File: 2981-find-longest-special-substring-that-occurs-thrice-i/2981-find-longest-special-substring-that-occurs-thrice-i.cpp
class Solution {
public:
    int maximumLength(string s) {
        vector<map<int, int>> freq(26);
        for(int i=0; i<s.length(); i++){
            char ch= s[i];
            int cnt=0;
            for(int j=i; j<s.length(); j++){
                if(ch==s[j]) cnt++;
                else break;
                freq[ch-'a'][cnt]++;
            }
        }
        int ans=-1;
        for(auto i: freq){
            for(auto itr= i.rbegin(); itr!= i.rend(); itr++){
                if(itr->second>=3){
                    ans= max(ans, itr->first);
                    break;
                }
            }
        }
        return ans;
        // for(auto i: freq){
        //     for(auto j: i){
        //         cout<<j.first<<" "<<j.second<<endl;
        //     }
        //     cout<<"--\n";
        // }
        //return 2;
    }
};

File: 2981-find-longest-special-substring-that-occurs-thrice-i/2981-find-longest-special-substring-that-occurs-thrice-i.java
class Solution {
    public int maximumLength(String s) {
        Map<Character, Map<Integer, Integer>> mp= new HashMap<>();
        for(int i=0; i<s.length(); i++){
            char ch= s.charAt(i);
            int cnt=0;
            for(int j=i; j<s.length(); j++){
                if(ch==s.charAt(j)) cnt++;
                else break;
               // mp.put(ch, mp.getOrDefault(ch, new HashMap<>()).put(cnt, mp.get(ch).getOrDefault(cnt,0))+1);
                mp.computeIfAbsent(ch, k -> new HashMap<>()).merge(cnt, 1, Integer::sum);
            }
        }
        int ans=-1;
        for(Map.Entry<Character, Map<Integer, Integer>> x: mp.entrySet()){
            for(Map.Entry<Integer, Integer> y: x.getValue().entrySet()){
                if(y.getValue()>=3){
                    ans = Math.max(ans, y.getKey());
                }
            }
        }
        return ans;
        
    }
}